packages/cli/internal/pkg/cli/util/timeout.go

```diff
@@ -0,0 +1,34 @@
+package util
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation/types"
+)
+
+var (
+	timeoutError = "the deployment is taking longer than expected. Current deployment status is %s. Please review the stack in CloudFormation"
+)
+
+func DeployWithTimeout(timeoutFunction func() error, deploymentStackStatus func() (types.StackStatus, error), timeoutDuration time.Duration) error {
```

Comment:
I'm worried this function is too specialized and doing too much to be a utility function - maybe make a simplified `runWithTimeout` utility function and wrap it as needed for the CDK deployments